src/zope/testing/doctestcase.py

```diff
@@ -0,0 +1,222 @@
+r"""Doctests in TestCase classes
+
+The original ``doctest`` unittest integration was based on
+``unittest`` test suites, which have fallen out of favor. This module
+provides a way to define doctests inside of unittest ``TestCase``
+classes. It also provides better integration with unittest test
+fixtures, because doctests use setup provided by the containing test
+case class.  It also provides access to unittest assertion
+methods.
+
+You can define doctests in 4 ways:
+
+- references to named files
+
+- strings
+
+- decorated functions with docstrings
+
+- reference to named files decorating test-specific setup functions
+
+.. some setup
+
+   >>> __name__ = 'tests'
+
+Here are some examples::
+
+    from zope.testing import doctestcase
+    import doctest
+    import unittest
+
+    g = 'global'
+    class MyTest(unittest.TestCase):
+
+        def setUp(self):
+            self.a = 1
+            self.globs = dict(c=9)
+
+        test1 = doctestcase.file('test1.txt', optionflags=doctest.ELLIPSIS)
+
+        test2 = doctestcase.docteststring('''
+          >>> self.a, g, c
+          (1, 'global', 9)
+        ''')
+
+        @doctestcase.doctestmethod(optionflags=doctest.ELLIPSIS)
+        def test3(self):
+            '''
+            >>> self.a, self.x, g, c
+            (1, 3, 'global', 9)
+            '''
+            self.x = 3
+
+        @doctestcase.doctestfile('test4.txt')
```

Comment:
Do we really need two different decorators `@doctestcase.file` and `@doctestcase.doctestfile`?